### PR TITLE
Saved card Security issue

### PR DESCRIPTION
**Pull request by MarijusCoding**
[Issue](https://invertus.atlassian.net/browse/SL-261)

### CODE REVIEW

File: src/Validation/CustomerCreditCardValidation.php [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    private $saferPayCardAliasRepository;
+
+    public function __construct(SaferPayCardAliasRepository $saferPayCardAliasRepository)
+    {
+        $this->saferPayCardAliasRepository = $saferPayCardAliasRepository;
+    }
+
+    /**
+     * @return true|Exception
+     * @throws UnauthenticatedCardUserException
```
Comment (MarijusCoding) [inline src/Validation/CustomerCreditCardValidation.php:48]: whitespace between

File: src/Validation/CustomerCreditCardValidation.php [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+     */
+    public function validate($idSavedCard, $idCustomer)
```
Comment (MarijusCoding) [inline src/Validation/CustomerCreditCardValidation.php:50]: if null given as $idSavedCard

File: src/Validation/CustomerCreditCardValidation.php [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+use Invertus\SaferPay\Exception\Restriction\UnauthenticatedCardUserException;
+use Invertus\SaferPay\Exception\SaferPayException;
+use Invertus\SaferPay\Logger\LoggerInterface;
+use Invertus\SaferPay\Repository\SaferPayCardAliasRepository;
+
+if (!defined('_PS_VERSION_')) {
+    exit;
+}
+
+class CustomerCreditCardValidation
```
Comment (zuk3975) [inline src/Validation/CustomerCreditCardValidation.php:36]:
```suggestion
class CustomerCreditCardValidat**or**
```

don't change now, but in future call services in suggested manner.

File: src/Validation/CustomerCreditCardValidation.php [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+                'id_customer' => $idCustomer
+            ]);
+
+            throw SaferPayException::unknownError();
+        }
+
+        $cardOwnerId = $this->saferPayCardAliasRepository->getCustomerIdByReferenceId(pSQL($idSavedCard), pSQL($idCustomer));
+
+        if (empty($cardOwnerId)) {
+            return throw UnauthenticatedCardUserException::unauthenticatedCard($cardOwnerId);
```
Comment (zuk3975) [inline src/Validation/CustomerCreditCardValidation.php:79]: Exceptions are thrown, not returned
```suggestion
            throw UnauthenticatedCardUserException::unauthenticatedCard($cardOwnerId);
```

**Pull request merged**